### PR TITLE
#7809: Edge: Add support for target="_blank" from the sidePanel

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Fixes https://github.com/pixiebrix/pixiebrix-extension/issues/7809

So yeah Edge can't handle  `target="_blank"`, which is set on every link via `base` tag:

https://github.com/pixiebrix/pixiebrix-extension/blob/1b1bab1ae9182955e95a49bab0c4f91a16bd82e9/src/sidebar/sidebar.html#L25-L26

## Testing

MV3 is broken so this PR cannot be _tested_ until #7831 is merged

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/sidebar/DefaultPanel.tsx [hunk 1]
```diff
@@ -50,29 +50,15 @@ const OnboardingContent: React.FunctionComponent = () => (
       <h4 className={styles.tinyCallout}>Need more help?</h4>
       <p>
         Visit the{" "}
-        <a
-          href="https://docs.pixiebrix.com/quick-start"
-          target="_blank"
```
Comment (fregante) [inline src/sidebar/DefaultPanel.tsx:55]: Since `sidebar.html` already has this attribute on all, we don't need to repeat it here

File: src/sidebar/Header.tsx [hunk 2]
```diff
@@ -98,8 +99,7 @@ const Header: React.FunctionComponent = () => {
         </Button>
       )}
       <Button
-        href="/options.html"
-        target="_blank"
+        href={getExtensionConsoleUrl()}
```
Comment (fregante) [inline src/sidebar/Header.tsx:102]: I also dropped this unusual usage, even if technically correct

**Comment by twschiller** [comment]
@fregante my understanding is that only the `/options.html` link was broken and that other links worked fine. Were you seeing something different?

File: src/sidebar/DefaultPanel.tsx (hunk 1 above)
Comment (grahamlangford) [reply src/sidebar/DefaultPanel.tsx:55]: We lose `rel=noopener noreferrer` though, don't we? Not really an issue with links we control, but still something to call out.

Comment (grahamlangford) [reply src/sidebar/DefaultPanel.tsx:55]: Also, don't we need to still set target for absolute urls?

Comment (twschiller) [reply src/sidebar/DefaultPanel.tsx:55]: > Also, don't we need to still set target for absolute urls?

See comment here, in my testing the absolute URLs were working OK: https://github.com/pixiebrix/pixiebrix-extension/pull/7832#issuecomment-1980827130

So before merging/adding a dependency, I think we should just test to see if MS Edge was mad about using the relative `/options.html` extension URL from the sidebar

**Comment by fregante** [comment]
I tried a plain link to Google directly in sidebar.html and it crashed the browser. I just assume it's safer to take control of the behavior for every link since this is such an egregious browser bug.

**Comment by grahamlangford** [comment]
If this is the route we're going to go, we should use https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/forbid-component-props.md to disallow target and rel on a tags.

**Review by twschiller — APPROVED** [review]

**Comment by fregante** [comment]
> disallow target and rel on a tags

Screenshot 5

✅ Added, just for documentation/cleaning purposes, but their existence is otherwise not a problem, it's only redundant.

File: src/sidebar/Tabs.tsx [hunk 3]
```diff
@@ -104,11 +104,8 @@ const TabWithDivider = ({
     >
       <Nav.Link
         {...props}
+        as="button"
```
Comment (fregante) [inline src/sidebar/Tabs.tsx:107]: The default for `Nav.Link`, unlike `Button`, is `href="#"`, which is awful (see the weird hack/comment below). 

We have a log of `Nav.Link` across the extension that would benefit from `as="button"` as well. Lint rule time?
